coord/src/rmr/rq.c

```diff
@@ -105,9 +107,21 @@ MRWorkQueue *RQ_New(int maxPending) {
   return q;
 }
 
-void RQ_Free(MRWorkQueue *q) {
+void RQ_Free(MRWorkQueue *q, RedisModuleCtx *ctx) {
   struct queueItem *req = NULL;
+  size_t pending_req = 0;
   while (NULL != (req = rqPop(q))) {
+    if (req->free_cb) {
+      req->free_cb(req->privdata);
+    } else {
+      ++pending_req;
+    }
+    if (pending_req) {
+      RedisModule_Log(ctx, "warning",
```

Comment:
I was running why you pass the `ctx`.. if it is just for logging you can log with static_ctx. No need to pass the ctx just for logging.

Reply:
Also, shouldn't we log it outside of the while? Maybe it is outside of the while and it hard for me to read it because of the cordcov warnings?

Reply:
yes just for logging
what's a static context? 🤔 

and yes it needs to be outside the loop thanks

Reply:
I tried to use `extern RedisModuleCtx *RSDummyContext; ` but the sanitizer detected a leak. I think it is released during shutdown before we get to releasing the queue.
Anyway i can pass NULL but the log prompt will include `<module>` instead of `<search>`